### ISSUE TITLE
Manual coercion via define_method (from mixin) doesn't work with Composition

### ISSUE THREAD

**Issue by ArthurN** [apotonick/reform#312]
Consider this composite form:

```ruby
  class CompositeForm < Reform::Form
    include Composition
    include Reform::Form::Coercion

    model :user

    property :birthdate, on: :profile

    validates :birthdate, presence: true

    def birthdate=(val)
      super(Chronic.parse(val).to_date)
    end
  end
```

(If you're not familiar, [Chronic](https://github.com/mojombo/chronic/) just converts Strings like "tomorrow" to actual Date/Time objects. I'm just using it here as an example; the particulars don't matter.)

and this simple test case:

```ruby
test do
    form = CompositeForm.new(user: OpenStruct.new, profile: OpenStruct.new)
    expect(form.validate({ birthdate: "12/03/1985" })).must_equal(true)  # for the sake of completeness

    form.save do |data|
      expect(data[:profile]['birthdate']).must_be_kind_of(Date)  # The real test
    end
end
```

All of that works great, as expected. However, if I move the manual coercion into a mixin and use `define_method` (or `class_eval` – same holds), all of a sudden the test fails (`Expected "12/03/1985" to be a kind of Date, not String.`). See [this gist](https://gist.github.com/ArthurN/e182bb609409e7878443).

I dug around Disposable's `Composition` class, and I suspect it has to do with the `expose_class` which is created at runtime via `Class.new`, but there is so much metaprogramming going on here I can't follow it all. Any thoughts?

**Comment by apotonick** [comment]
Why do you `include Reform::Form::Coercion`, can you please comment that line to make sure it's not that line? Since you're doing your coercion manually, there's no point in that. :wink:

**Comment by apotonick** [comment]
It's odd, look, this works.

```ruby
  class TwinCompositionDefineMethod < Disposable::Twin
    include Composition

    property :id, on: :song

    define_method :id do
      super() + 9
    end
  end

  it do
    twin = TwinCompositionDefineMethod.new(song: song)
    twin.id.must_equal 9
  end
```

**Comment by apotonick** [comment]
This is fixed in Reform 2.1.1.
